eng/scripts/automation_init.sh

```diff
@@ -1,25 +1,58 @@
+#!/bin/bash
+
+if [ -z $1 ]; then
+    echo "Please input outputfile"
+    exit 1
+fi
+echo $1
+
 set -x
 set -e
+sudo npm install -g autorest@latest && sudo npm install -g typescript
+
+TMPDIR="/tmp"
 if [ ! "$(go version | awk '{print $3}' | cut -c 3-6)" = "1.16" ]
 then
   wget https://golang.org/dl/go1.16.9.linux-amd64.tar.gz
   tar -C $TMPDIR -xzf go1.16.9.linux-amd64.tar.gz
-  export PATH=$TMPDIR/go/bin:$PATH
+  export GOROOT=$TMPDIR/go
+  export PATH=$GOROOT/bin:$PATH
 fi
-export GOPATH="$(realpath ../../../..)"
-if [ ! -d "$GOPATH/bin" ]
-then
-  mkdir $GOPATH/bin
+
+DIRECTORY=$(cd `dirname $0` && pwd)
+
+if [ "$GOPATH" == "" ]; then
+  WORKFOLDER="$(realpath $DIRECTORY/../../../)"
+  echo $WORKFOLDER
+  export GOPATH=$WORKFOLDER/gofolder
 fi
-PATH=$PATH:$GOPATH/bin
+
+if [ ! -d "$GOPATH/bin" ]; then
+  echo "create gopath folder"
+  mkdir -p $GOPATH/bin
+fi
+echo $GOPATH
+
 export GO111MODULE=on
-cd eng/tools/generator && go build && cp generator $GOPATH/bin && cd ../../..
-ln -s /usr/bin/pwsh $GOPATH/bin/pwsh.exe
-cat > $2 << EOF
+
+generatorDirectory="$(realpath $DIRECTORY/../tools/generator)"
+cd $generatorDirectory
+go build
+
+cp generator $GOPATH/bin/
+export PATH=$GOPATH/bin:$PATH
+cd $DIRECTORY
+
+if [ ! -f "$GOPATH/bin/pwsh.exe" ]; then
+  ln -s /usr/bin/pwsh $GOPATH/bin/pwsh.exe
+fi
+
+cat > $1 << EOF
 {
   "envs": {
-    "PATH": "$PATH:$GOPATH",
-    "GOPATH": "$GOPATH"
+    "PATH": "$GOPATH:$PATH",
+    "GOPATH": "$GOPATH",
+    "GOROOT": "$GOROOT"
   }
 }
-EOF
+EOF
```

Comment:
Have we tested it with the existed SDK automation? I'm afraid some env are different under two framework which may cause some unexpected failure.